extensions/security/runtime-spi/src/main/java/io/quarkus/security/spi/runtime/SecurityEvent.java

```diff
@@ -0,0 +1,33 @@
+package io.quarkus.security.spi.runtime;
+
+import java.util.Map;
+
+import io.quarkus.security.identity.SecurityIdentity;
+
+/**
+ * Common interface for all CDI security events, which allows to consume all the events.
+ */
+public interface SecurityEvent {
+
+    SecurityEvent NOOP = new SecurityEvent() {
```

Comment:
What is the idea behind this `NOOP` ?

Reply:
In order to check with `BeanManager` whether event type is observed, I need actual instance of that event. And ever bean that fire security events need this instances, so I thought it would be nice not to create them over and over again.

Reply:
Same goes for empty instances on security event implementations.